pages/acceptInvite.page.tsx

```diff
@@ -0,0 +1,132 @@
+  url: string;
+  inviteType: string;
+  id: string;
+  code: string;
+}
+
+export const fetchAcceptInvite = ({
+  apiToken,
+  url,
+  inviteType,
+  id,
+  code,
+}: FetchAcceptInviteProps): Promise<Response> => {
+  return fetch(process.env.REST_API_URL + url, {
+    method: 'PUT',
+    headers: {
+      authorization: `Bearer ${apiToken}`,
+      'content-type': 'application/vnd.api+json',
+      accept: 'application/vnd.api+json',
+    },
+    body: JSON.stringify({
+      data: {
+        type: inviteType,
+        id,
+        attributes: { code },
+      },
+    }),
+  });
+};
+
+const AcceptInvitePage = (): ReactElement => {
+  const { t } = useTranslation();
+  const { appName } = useGetAppSettings();
+  const { enqueueSnackbar } = useSnackbar();
+  const router = useRouter();
+  const session = useRequiredSession();
+  const currentUserAccountListId = useAccountListId();
+
+  // Ref to track if the API call has already been triggered
+  const hasFetchedRef = useRef(false);
+
+  useEffect(() => {
+    if (!router.isReady || hasFetchedRef.current) {
+      return;
+    }
+    const { inviteCode, accountInviteId, orgInviteId, orgId } = router.query;
+    const inviterAccountListId = router.query.accountListId || undefined;
+    const acceptInvite = async (id, code, url) => {
+      const inviteType = url.includes('organizations')
+        ? 'organization_invites'
+        : 'account_list_invites';
+
+      try {
+        const apiToken = session?.apiToken;
+
+        const response = await fetchAcceptInvite({
+          apiToken,
+          url,
+          inviteType,
+          id,
+          code,
+        });
+
+        if (!response.ok) {
+          throw new Error('Network response was not ok');
+        }
+        if (url.includes('organizations')) {
+          enqueueSnackbar(t('Accepted invite successfully.'), {
+            variant: 'success',
+          });
+          router.push(
+            `/accountLists/${currentUserAccountListId}/settings/preferences`,
+          );
+        } else {
+          enqueueSnackbar(t('Accepted invite successfully.'), {
+            variant: 'success',
+          });
+          router.push(`/accountLists/${currentUserAccountListId}`);
+        }
+      } catch (err) {
+        enqueueSnackbar(
+          t(
+            'Unable to accept invite. Try asking the account holder to resend the invite.',
+          ),
+          {
+            variant: 'error',
+          },
+        );
+      }
+    };
+
+    if (accountInviteId && inviteCode && inviterAccountListId) {
+      const url = `account_lists/${inviterAccountListId}/invites/${accountInviteId}/accept`;
+      acceptInvite(accountInviteId, inviteCode, url);
+      hasFetchedRef.current = true;
+    } else if (orgInviteId && inviteCode && orgId) {
+      const url = `organizations/${orgId}/invites/${orgInviteId}/accept`;
+      acceptInvite(orgInviteId, inviteCode, url);
+      hasFetchedRef.current = true;
+    }
+  }, [router.query]);
```

Comment:
Whst if they have none of these query parameters, should we add a warning snackbar to warn the users they don't have the correct paramters?

Reply:
Angular shows an error toast and redirects to the dashboard

https://github.com/CruGlobal/mpdx_web/blob/f02ea571631feb3ff6f52e17f454d4c4af3f2896/src/acceptInvite/acceptInvite.component.ts#L107-L111